source/extensions/filters/common/rbac/matchers.h

```diff
@@ -163,6 +163,19 @@ class PortMatcher : public Matcher {
   const uint32_t port_;
 };
 
+class PortRangeMatcher : public Matcher {
+public:
+  PortRangeMatcher(const ::envoy::type::v3::Int32Range& range)
+      : start_(range.start()), end_(range.end()) {}
```

Comment:
Arguably we should reject negative and OOB values here for ports.

Reply:
To validate the range of ports, we are better to add a validation rule in RBAC proto. 

https://github.com/envoyproxy/envoy/blob/f97bdf143f570e10bd919369235fc801f3a65fd2/api/envoy/config/rbac/v3/rbac.proto#L189-L190

But the type `type.v3.Int32Range` does not have any validation rules, and only certain types have validation rules that is located in [validate.proto](https://github.com/envoyproxy/protoc-gen-validate/blob/main/validate/validate.proto#L36-L66), which belongs to another repo [envoyproxy/protoc-gen-validate](https://github.com/envoyproxy/protoc-gen-validate). As a result, we are not easy to reject invalid or illegal values.

In addition, `type.v3.Int32Range` is also used in destination port matching in listener filter, but the `ListenerFilterDstPortMatcher` does not check its input.

https://github.com/envoyproxy/envoy/blob/f97bdf143f570e10bd919369235fc801f3a65fd2/api/envoy/config/listener/v3/listener_components.proto#L330-L332

https://github.com/envoyproxy/envoy/blob/f97bdf143f570e10bd919369235fc801f3a65fd2/source/common/network/filter_matcher.h#L40-L46

So I think maybe we do not have to add a check?

Reply:
Yes, this is a limit of PGV annotations and proto typing, I don't think it can be done cleanly at the proto annotation level. I think both `ListenerFilterDstPortMatcher` and your code could share a common utility that at config ingestion time, checks the port range and throws if invalid; this is the next best thing.